ledger-service/db-backend/src/main/scala/com/digitalasset/http/dbbackend/Queries.scala

```diff
@@ -723,7 +774,7 @@ private final class PostgresQueries(tablePrefix: String, tpIdCacheMaxEntries: Lo
       s"""
         INSERT INTO $contractTableNameRaw
         VALUES (?, ?, ?::jsonb, ?, ?::jsonb, ?, ?, ?)
-        ON CONFLICT (contract_id) DO NOTHING
+        ON CONFLICT (contract_id, tpid) DO NOTHING
```

Comment:
Is there an Oracle version for do nothing on conflict? (just curious)

Reply:
Yes, and I recommend sitting down before [clicking on this link](https://github.com/digital-asset/daml/pull/9286#discussion_r604141044).

Reply:
And regarding the followup d50e483a88f9edb729571e77d30ef82560388a60 I just added to this PR, I don't know why Oracle accepted it without this change.

Reply:
🤷‍♂️